internal/runners/publish/publish.go

```diff
@@ -143,12 +143,16 @@ func (r *Runner) Run(params *Params) error {
 	isRevision := false
 	if params.Version != "" {
 		// Attempt to get the version if it already exists, it not existing is not an error though
-		i, err := model.GetIngredientByNameAndVersion(params.Namespace, params.Name, params.Version)
+		i, err := model.GetIngredientByNameAndVersion(reqVars.Namespace, reqVars.Name, params.Version)
```

Comment:
Note this also fixes another bug I found; this was still using the namespace and name from the user input, which might be empty.